### PR TITLE
Image Text Block

### PR DESCRIPTION
**Pull request by gabriel-glo**
Modify and update existing components
Add Image Text block attributes, options, editor components, styling and frontend template.

### CODE REVIEW

File: wp-content/themes/eighshift-task/src/blocks/components/heading/components/heading-editor.js [hunk 1]
```diff
@@ -7,7 +7,6 @@ export const HeadingEditor = (props) => {
     blockClass,
     content,
     onChangeContent,
-    level,
```
Comment (iruzevic) [inline wp-content/themes/eighshift-task/src/blocks/components/heading/components/heading-editor.js]: why did you remove level?

File: wp-content/themes/eighshift-task/src/blocks/custom/image-text/components/image-text-editor.js [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+      btnTitle,
+      btnStyleSize,
+      btnStyleColor,
+      btnStyleSizeWidth,
+    },
+    actions,
+  } = props;
+
+  const blockClasses = classnames([
+    `${blockClass}__position--${mediaPosition}`,
```
Comment (iruzevic) [inline wp-content/themes/eighshift-task/src/blocks/custom/image-text/components/image-text-editor.js]: position  of what?

you are saying that position is left, but of what, image or text?
a better selector would be example: `${blockClass}__image-position--${mediaPosition}`

File: wp-content/themes/eighshift-task/src/blocks/custom/image-text/components/image-text-options.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+          options={[
+            { label: __('Left', 'eighshift-task'), value: 'left' },
+            { label: __('Right', 'eighshift-task'), value: 'right' },
+          ]}
+          onChange={onChangeMediaPosition}
+        />
+      }
+
+      {onChangeIncludeBtn &&
+        <CheckboxControl
```
Comment (iruzevic) [inline wp-content/themes/eighshift-task/src/blocks/custom/image-text/components/image-text-options.js]: try implementing https://github.com/WordPress/gutenberg/tree/master/packages/components/src/toggle-control it looks better and is more UX friendly

File: wp-content/themes/eighshift-task/src/blocks/custom/image-text/image-text-style.scss [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+    #{$this}__visual {
+      order: 1;
+    }
+
+    #{$this}__textual {
+      margin-bottom: map-get-strict($block-image-text, vertical-spacing);
+
+      @include media(tablet up) {
+        margin-bottom: 0;
+        margin-right: calc-grid-width(1);
```
Comment (iruzevic) [inline wp-content/themes/eighshift-task/src/blocks/custom/image-text/image-text-style.scss]: if you put this as a padding, whan you dont need to fiddle with grid columns. just put columns equal size 6 and 6. Simpler to maintain

File: wp-content/themes/eighshift-task/src/blocks/custom/image-text/image-text-style.scss [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+      }
+    }
+  }
+
+  .heading {
```
Comment (iruzevic) [inline wp-content/themes/eighshift-task/src/blocks/custom/image-text/image-text-style.scss]: why not bem?

File: wp-content/themes/eighshift-task/src/blocks/custom/image-text/image-text-style.scss [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+    padding-bottom: 25px;
+  }
+
+  .paragraph {
```
Comment (iruzevic) [inline wp-content/themes/eighshift-task/src/blocks/custom/image-text/image-text-style.scss]: why not bem?

File: wp-content/themes/eighshift-task/src/blocks/custom/image-text/image-text.js [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+        mediaId: value.id,
+        mediaUrl: value.url,
+      });
+    },
+  };
+
+  return (
+    <Fragment>
+      <InspectorControls>
+        <ImageTextOptions
```
Comment (iruzevic) [inline wp-content/themes/eighshift-task/src/blocks/custom/image-text/image-text.js]: this is a block component just pass all the attributes and actions.

File: wp-content/themes/eighshift-task/src/blocks/components/heading/components/heading-editor.js (hunk 1 above)
Comment (gabriel-glo) [reply wp-content/themes/eighshift-task/src/blocks/components/heading/components/heading-editor.js]: My bad. Probably because linter was throwing warnings.

File: wp-content/themes/eighshift-task/src/blocks/custom/image-text/image-text-style.scss (hunk 4 above)
Comment (gabriel-glo) [reply wp-content/themes/eighshift-task/src/blocks/custom/image-text/image-text-style.scss]: But then I lose the required spacing, because with padding property I'm overriding the columns gutter padding.

Comment (iruzevic) [reply wp-content/themes/eighshift-task/src/blocks/custom/image-text/image-text-style.scss]: 👍

File: wp-content/themes/eighshift-task/src/blocks/custom/image-text/image-text.js [hunk 8]
```diff
@@ -0,0 +1,107 @@
...
+import { ParagraphOptions } from '../../components/paragraph/components/paragraph-options';
+import { ButtonOptions } from '../../components/button/components/button-options';
+
+import { HeadingToolbar } from '../../components/heading/components/heading-toolbar';
+import { ParagraphToolbar } from '../../components/paragraph/components/paragraph-toolbar';
+
+export const ImageText = (props) => {
+  const {
+    setAttributes,
+    attributes: {
```
Comment (iruzevic) [inline wp-content/themes/eighshift-task/src/blocks/custom/image-text/image-text.js]: if you desctructure additional key here as attributes. then you can pass only attributes key in ImageTextOptions

File: wp-content/themes/eighshift-task/src/blocks/custom/image-text/image-text.js (hunk 7 above)
Comment (iruzevic) [reply wp-content/themes/eighshift-task/src/blocks/custom/image-text/image-text.js]: comment before in attributes pass only attributes like this.

attributes={attributes}

File: wp-content/themes/eighshift-task/src/blocks/custom/image-text/components/image-text-options.js [hunk 9]
```diff
@@ -1,12 +1,13 @@
...
   const {
-    mediaPosition,
-    onChangeMediaPosition,
-    includeBtn,
-    onChangeIncludeBtn,
+    attributes: {
+      mediaPosition,
+      includeBtn,
+    },
+    actions,
```
Comment (iruzevic) [inline wp-content/themes/eighshift-task/src/blocks/custom/image-text/components/image-text-options.js]: distracture actions also and use only props values like

onChange={onChangeMediaPosition}
